src/app/api/[[...route]]/routes/sharehouse.route.ts

```diff
@@ -172,9 +181,200 @@ app.delete('/:shareHouseId', async (c) => {
   }
 });
 
-app.post('/:landlordId', (c) => {
-  const landlordId = c.req.param('landlordId');
-  return c.json({
-    message: `Landlord id for creating new share hose: ${landlordId}`,
-  });
+app.post('/', zValidator('json', shareHouseCreationSchema), async (c) => {
+  try {
+    const session = await auth();
+
+    if (!session) {
+      return c.json({
+        message: 'You are not logged in!',
+      });
+    }
+    const landlordId = session.user.id;
+    const data = c.req.valid('json');
+
+    const landlord = await prisma.landlord.findUnique({
+      where: {
+        id: landlordId,
+      },
+      include: {
+        shareHouses: true,
+      },
+    });
+
+    if (!landlord)
+      return c.json(
+        { error: 'Landlord not found for the given landlordId' },
+        404,
+      );
+
+    // Check for existing shareHouse by name
+    const existingShareHouse = await prisma.shareHouse.findFirst({
+      where: {
+        name: data.name,
+      },
+    });
```

Comment:
With your implementation, this checks every single sharehouse in the table regardless of the landlords. Therefore, could you target only the sharehouses that the landlord owns and check whether there's the same name?

Reply:
I fixed the if statement to pass the shareHouse with the same name if the landlord differs.